gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiServiceCockpitImpl.java

```
@@ -34,47 +39,147 @@
+        final SwaggerApiEntity api = swaggerService.createAPI(swaggerDescriptor, DefinitionVersion.V2);
+        api.setPaths(null);
+
+        return this.apiService.updateFromSwagger(apiId, api, swaggerDescriptor);
+    }
+
+    private ApiEntity createApiEntity(String apiId, String userId, ImportSwaggerDescriptorEntity swaggerDescriptor) {
+        final SwaggerApiEntity api = swaggerService.createAPI(swaggerDescriptor, DefinitionVersion.V2);
+        api.setPaths(null);
+
         final ObjectNode apiDefinition = objectMapper.valueToTree(api);
         apiDefinition.put("id", apiId);
 
         final ApiEntity createdApi = apiService.createWithApiDefinition(api, userId, apiDefinition);
-        apiService.createSystemFolder(createdApi.getId());
-        apiService.createOrUpdateDocumentation(swaggerDescriptor, createdApi, true);
-        apiService.createMetadata(api.getMetadata(), createdApi.getId());
-
+        pageService.createAsideFolder(apiId, GraviteeContext.getCurrentEnvironment());
+        pageService.createOrUpdateSwaggerPage(apiId, swaggerDescriptor, true);
+        apiMetadataService.create(api.getMetadata(), createdApi.getId());
         return createdApi;
     }
+
+    private NewPlanEntity createKeylessPlan(String apiId, String environmentId) {
+        NewPlanEntity plan = new NewPlanEntity();
+        plan.setId(UuidString.generateForEnvironment(environmentId, apiId));
+        plan.setName("Keyless plan");
+        plan.setSecurity(PlanSecurityType.KEY_LESS);
+        plan.setApi(apiId);
+        plan.setStatus(PlanStatus.PUBLISHED);
+
+        return plan;
+    }
+
+    private void publishSwaggerDocumentation(String apiId) {
+        List<PageEntity> apiDocs = pageService.search(
+            new PageQuery.Builder().api(apiId).type(PageType.SWAGGER).build(),
+            GraviteeContext.getCurrentEnvironment()
+        );
+
+        if (apiDocs.size() == 0) {
+            logger.error("No swagger documentation to publish");
```

Improvments suggestions :
- use isEmpty instead of size == 0
- throw an exception instead of just logging an error, as the following get(0) will fail